parsers/US_PREPA.py

```diff
@@ -0,0 +1,225 @@
+    provided, we should default it to now. If past data is not available,
+    raise a NotImplementedError. Beware that the provided target_datetime is
+    UTC. To convert to local timezone, you can use
+    `target_datetime = arrow.get(target_datetime).to('America/New_York')`.
+    Note that `arrow.get(None)` returns UTC now.
+  logger: an instance of a `logging.Logger` that will be passed by the
+    backend. Information logged will be publicly available so that correct
+    execution of the logger can be checked. All Exceptions will automatically
+    be logged, so when something's wrong, simply raise an Exception (with an
+    explicit text). Use `logger.warning` or `logger.info` for information
+    that can useful to check if the parser is working correctly. A default
+    logger is used so that logger output can be seen when coding / debugging.
+  Returns:
+  --------
+  If no data can be fetched, any falsy value (None, [], False) will be
+    ignored by the backend. If there is no data because the source may have
+    changed or is not available, raise an Exception.
+  A dictionary in the form:
+  {
+    'zoneKey': 'FR',
+    'datetime': '2017-01-01T00:00:00Z',
+    'production': {
+        'biomass': 0.0,
+        'coal': 0.0,
+        'gas': 0.0,
+        'hydro': 0.0,
+        'nuclear': null,
+        'oil': 0.0,
+        'solar': 0.0,
+        'wind': 0.0,
+        'geothermal': 0.0,
+        'unknown': 0.0
+    },
+    'storage': {
+        'hydro': -10.0,
+    },
+    'source': 'mysource.com'
+  }
+  """
+
+  global renewable_output
+
+  if target_datetime is not None:
+      raise NotImplementedError('The datasource currently implemented is only real time')
+
+  r = session or requests.session()
+
+  data = { #To be returned as response data
+    'zoneKey': zone_key,
+    #'datetime': '2017-01-01T00:00:00Z',
+    'production': {
+        'biomass': 0.0,
+        'coal': 0.0,
+        'gas': 0.0,
+        'hydro': 0.0,
+        'nuclear': 0.0,
+        'oil': 0.0,
+        'solar': 0.0,
+        'wind': 0.0,
+        'geothermal': 0.0,
+        'unknown': 0.0
+    },
+#   'storage': {
+#        'hydro': -10.0,
+#    },
+    'source': 'aeepr.com'
+  }
+
+  renewable_output = 0.0 #Temporarily stored here. We'll subtract solar, wind and biomass (landfill gas) from it and assume the remainder, if any,  is hydro
+
+  #Step 1: fetch production by generation type
+  #Note: seems to be rounded down (to an integer)
+  #Total at the top of the page fetched in step 3 isn't rounded down, but seems to be lagging behind sometimes.
+  #Difference is only minor, so for now we will NOT try to parse the total and add the difference to "unknown"
+  res = r.get("https://aeepr.com/es-pr/generacion/Documents/combustibles.aspx")
+
+  assert res.status_code == 200, 'Exception when fetching production for ' \
+                               '{}: error when calling url={}'.format(
+                                   zone_key, url)
```

Comment:
This will error because `url` is not defined.

Reply:
fixed this and the other two places where this occurs